### PR TITLE
Themes: Split main into single-site, multi-site,and logged-out

### LINKED ISSUE #2714: Themes: Split `main` into `single-site`, `multi-site`, and `logged-out`
I think part of the cognitive burden and mess of `main.jsx` is the number of conditionals and ternaries that check if we're dealing with any of those modes. To relieve that, I think that in addition to #2444, we should split it up into at least those three. Each will then only need a limited subset of those components we're using right now -- e.g. `single-site` won't need `ThemesSiteSelectorModal` and the `CurrentTheme` bar, while the other ones won't need any Jetpack related error message components.

We might then also iterate on `theme-options`, and remove the `isHidden` fields (of #2444), instead `pick`ing the options we need individually in each of those, uh, view modes.

### PR DESCRIPTION
**Pull request by ockham**
`my-sites/themes/main.jsx` has been overflowing with imports and methods, most of which were used in only one or two out of the single-site, multi-site, and logged-out scenario. Likewise, `theme-options` had to cater for all of those three cases.

This PR hence breaks `main` into 3 components corresponding to each of those cases and moves `getButtonOptions()` to each of them, removing conditionals that would only serve the other cases. Because of fewer imports, methods, and branches in each of those files, I believe they are now more legible. It's however even more important now to factor out re-usable parts that at least two of those components use. Hence, this PR also introduces `<ThemePreview>`.

Also, `logged-out.jsx` will hopefully come in handy for SSR.

Fixes #2714 

PS:
* `controller.js` could obviously still need a makeover after this PR, but since we might end up with a slightly different routing anyway, I'm postponing that.
* Historically, we introduced `themes-selection` to bundle `themes-list` and search for SSR. That might be an obsolete level of abstraction with the new `logged-out` component (which also has theme preview) that this PR introduces, so in another iteration, we are probably going to remove `themes-selection`.

### CODE REVIEW

File: client/my-sites/themes/theme-preview.jsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+		const previewUrl = getPreviewUrl( this.props.theme );
+
+		return(
+      <WebPreview showPreview={ this.props.showPreview }
+        onClose={ this.props.onClose }
+        previewUrl={ previewUrl } >
+        <Button primary
+          onClick={ this.onButtonClick }
+          >{ this.props.buttonLabel }</Button>
+      </WebPreview>
```
Comment (ehg) [inline client/my-sites/themes/theme-preview.jsx]: Please indent me.

File: client/my-sites/themes/theme-options.js [hunk 2]
```diff
@@ -3,70 +3,11 @@
  */
 import assign from 'lodash/object/assign';
 import mapValues from 'lodash/object/mapValues';
-import pick from 'lodash/object/pick';
-import merge from 'lodash/object/merge';
 
 /**
  * Internal dependencies
  */
 import Helper from 'lib/themes/helpers';
```
Comment (ehg) [inline client/my-sites/themes/theme-options.js]: The functions in this file seem a bit lonely. Maybe they could be moved to `my-sites/themes/helpers`, also I wonder if we should move `lib/themes/helpers` to the same file.

File: client/my-sites/themes/theme-preview.jsx (hunk 1 above)
Comment (ockham) [inline client/my-sites/themes/theme-preview.jsx]: No component left behind: 7c1ef85

File: client/my-sites/themes/theme-options.js (hunk 2 above)
Comment (ockham) [inline client/my-sites/themes/theme-options.js]: > The functions in this file seem a bit lonely.

Agreed.

> Maybe they could be moved to `my-sites/themes/helpers`, also I wonder if we should move `lib/themes/helpers` to the same file.

There's a couple of occurrences outside `my-sites/themes` and even outside `my-sites` that require that file (some related to signup). Okay to move everything from `theme-options` to `lib/themes` (at least for now)?

Comment (ockham) [inline client/my-sites/themes/theme-options.js]: Moved `addTracking()` and friends to `lib/themes/helpers`: 38d8ea3.

**Comment by ehg** [comment]
Looks good! Can't see anything obviously wrong with the code, and tests of LO/SS/MS/analytics seem fine. Going through the code, I noticed that the separation of concerns is a boon to readability. :+1:

**Pull request merged**